SpringData/src/main/java/com/epam/jmp2/service/impl/AccidentDBServiceImpl.java

```diff
@@ -1,75 +1,116 @@
-        return null;
-
-    }
-    public Boolean update(RoadAccident roadAccident) {
-        // To be filled by mentee
-        return null;
-    }
-
-	public DistrictAuthorityRepository getDistrictAuthorityRepository() {
-		return districtAuthorityRepository;
-	}
-
-	public void setDistrictAuthorityRepository(DistrictAuthorityRepository districtAuthorityRepository) {
-		this.districtAuthorityRepository = districtAuthorityRepository;
-	}
+package com.epam.jmp2.service.impl;
+
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+
+
+
+import com.epam.jmp2.dbrepositories.AccidentRepository;
+import com.epam.jmp2.dbrepositories.DistrictAuthorityRepository;
+import com.epam.jmp2.entities.Accident;
+import com.epam.jmp2.entities.RoadSurfaceCondition;
+import com.epam.jmp2.entities.WeatherCondition;
+import com.epam.jmp2.model.RoadAccident;
+import com.epam.jmp2.model.RoadAccidentBuilder;
+import com.epam.jmp2.model.TimeOfDay;
+import com.epam.jmp2.service.AccidentService;
+import com.epam.jmp2.util.Util;
+
+@Component
+public class AccidentDBServiceImpl implements AccidentService {
+
+    @Autowired
+    private AccidentRepository accidentRepository;
+    
+    @Autowired
+    private DistrictAuthorityRepository districtAuthorityRepository;
+
+    public Accident findOne(String accidentId) {
+        return accidentRepository.findOne(accidentId);
+    }
+    
+    public Iterable<RoadAccident> getAllAccidentsByDate(String date) {
+    	List<Accident> accidentByWeatherCondition = accidentRepository.findByDate(date);
+        return getRoadAccidentFromAccident(accidentByWeatherCondition);
+    }
+
+    public Iterable<RoadAccident> getAllAccidentsByRoadCondition(Integer label) {
+    	RoadSurfaceCondition roadSurfaceCondition = new RoadSurfaceCondition();
+    	roadSurfaceCondition.setCode(label);
+    	List<Accident> accidentByRoadSurfaceCondition = accidentRepository.findByRoadSurfaceCondition(roadSurfaceCondition);
+    	return getRoadAccidentFromAccident(accidentByRoadSurfaceCondition);
+    }
+    
+	public Long getAllAccidentsByWeatherConditionAndYear(Integer weatherCondition, String year) throws ParseException {
+    	WeatherCondition weatherConditionItem = new WeatherCondition();
+    	weatherConditionItem.setCode(weatherCondition);
+    	
+        List<Accident> accidentList = accidentRepository.findAccidentsByWeatherCondition(weatherConditionItem);
+        List<Accident> yearFromDate = getYear(accidentList);
+        
+		Map<String, Long> result = (yearFromDate).stream().collect(Collectors.groupingBy(Accident::getYear, Collectors.counting()));
+        return result.get(year);
+    }
+
+    public Iterable<RoadAccident> updateTimePeriod(String date) {
```

Comment:
You should have two methods here 
getAllAccidentsByDate(Date date)
update(RoadAccident roadAccident)

I am not sure why have you merged this two methods into a single method.